zebra/zapi_msg.c

```diff
@@ -957,17 +957,16 @@ int zsend_pw_update(struct zserv *client, struct zebra_pw *pw)
 
 /* Send response to a get label chunk request to client */
 int zsend_assign_label_chunk_response(struct zserv *client, vrf_id_t vrf_id,
-				      uint8_t proto, uint16_t instance,
 				      struct label_manager_chunk *lmc)
 {
 	int ret;
 	struct stream *s = stream_new(ZEBRA_MAX_PACKET_SIZ);
 
 	zclient_create_header(s, ZEBRA_GET_LABEL_CHUNK, vrf_id);
 	/* proto */
-	stream_putc(s, proto);
+	stream_putc(s, client->proto);
 	/* instance */
-	stream_putw(s, instance);
+	stream_putw(s, client->instance);
 
```

Comment:
Perhaps we could also add session_id parameter in the message. This way, the client could check that the answer corresponds to the correct session.

Reply:
So ... this is, I think, a bit of legacy left over from the days of the infamous label-manager proxy. Other messages don't carry this extra baggage around - unless tcp goes horribly wrong, this is the right connection. I'd prefer not to keep adding more stuff to these label-manager messages.